plaidml/bridge/openvino/ops/roi_pooling.cpp

```diff
@@ -0,0 +1,153 @@
+
+      // get max value from bin, then put it to pooled tensor.
+      std::vector<edsl::TensorDim> dims(crop.rank());
+      crop.bind_dims(dims);
+      std::vector<edsl::TensorIndex> idx(crop.rank());
+      auto bin_max = edsl::Contraction()
+                         .outShape({dims[0], dims[1], edsl::TensorDim(pooled_h), edsl::TensorDim(pooled_w)})
+                         .outAccess({idx[0], idx[1], edsl::TensorIndex(i), edsl::TensorIndex(j)})
+                         .max(crop(idx));
+      pooled_tensor = (i == 0) ? bin_max : edsl::select(pooled_tensor > bin_max, pooled_tensor, bin_max);
+    }
+  }
+  return pooled_tensor;
+}
+
+edsl::Tensor bilinear_pooling(edsl::Tensor I, std::vector<float>& coord, int64_t pooled_h, int64_t pooled_w) {
+  auto x_1 = coord[0];
+  auto y_1 = coord[1];
+  auto x_2 = coord[2];
+  auto y_2 = coord[3];
+
+  auto roi_width = y_2 - y_1;
+  auto roi_height = x_2 - x_1;
+
+  auto roi_h_scale = roi_height / pooled_h;
+  auto roi_w_scale = roi_width / pooled_w;
+
+  auto h_tensor = edsl::cast(edsl::index({edsl::TensorDim(pooled_h)}, 0), DType::FLOAT32) * roi_h_scale + x_1;
+  auto w_tensor = edsl::cast(edsl::index({edsl::TensorDim(pooled_w)}, 0), DType::FLOAT32) * roi_w_scale + y_1;
+
+  auto gather_w = edsl::gather(I, w_tensor).axis(3).interpolationMode(edsl::InterpolationMode::LINEAR);
+  auto gather_h = edsl::gather(gather_w, h_tensor).axis(2).interpolationMode(edsl::InterpolationMode::LINEAR);
+
+  return gather_h;
+}
+
+}  // namespace
+
+namespace PlaidMLPlugin {
+
+void registerROIPooling() {
+  registerOp("ROIPooling", [](const Context& ctx) {
+    auto* layer = ngraph::as_type<ngraph::opset4::ROIPooling>(ctx.layer);
+    IE_ASSERT(ctx.operands.size() == 2);
+    auto I = ctx.operands.at(0);
+
+    auto pooled_shape = layer->get_output_size();
+    auto pooled_height = static_cast<int64_t>(pooled_shape[0]);
+    auto pooled_width = static_cast<int64_t>(pooled_shape[1]);
+    auto spatial_ratio = layer->get_spatial_scale();
+    auto method = layer->get_method();
+
+    const static int BOX_ELEMENT_SIZE = 5;
+    auto coords_box = cast_constant_operand<float>(1, layer);
+    IE_ASSERT((coords_box.size() % BOX_ELEMENT_SIZE) == 0);
+
+    int height = 0, width = 0;
+    if (method == "bilinear") {
+      auto shapes = I.compute_shape().sizes();
+      height = shapes[2];
+      width = shapes[3];
+    }
+
+    std::vector<edsl::Tensor> ROI_pools;
+    // 2D input tensor of shape [NUM_ROIS, 5] describing box
+    for (auto index = coords_box.begin(); index != coords_box.end(); index += BOX_ELEMENT_SIZE) {
+      // consisting of 5 element tuples: [batch_id, x_1, y_1, x_2, y_2]
+      auto batch_id = *index;
+      std::vector<float> coord(index + 1, index + BOX_ELEMENT_SIZE);
+
+      auto batch_indices = edsl::index({edsl::TensorDim(1)}, 0) + batch_id;
+      auto slice_I = edsl::gather(I, batch_indices).axis(0);
+
+      edsl::Tensor pooled_tensor;
+      if (method != "bilinear") {
```

Comment:
I think it is better to use switch here, and throws an error if the method is neither bilinear or max. They may add more methods like avg or something else in the future.

Reply:
thanks for comment, I try to use switch here, but we only can get method as string (from `layer->get_method()`), it means we have to build enum class or map in this file for using switch. 
According to latest openvino upstream, it create `ROIPoolingType` in `ngraph::helper`, then, use this `ROIPoolingType` in test instance, but still translate this `ROIPoolingType` to string in `setup` step. it means` ngraph::opset4::ROIPooling`  always carry `method` as string, so I think maybe it's better we always use method as string in this file. 
I change this code like  `if (method == "max") {...... } else if (method == "bilinear") {..... }else  {THROW_IE_EXCEPTION}`. I'm not sure if this is acceptable?

Reply:
Yes, I think it is good and is ready to submit to plaidml.